chanzuckerberg/edu-design-system
# chore: add linter for tier 1 color css variables

### Summary:
Ticket: https://czi-tech.atlassian.net/browse/EDS-707

For the purposes of theming, we only want to use tier 2 and tier 3 color CSS variables in component styling in EDS (with some rare exceptions). This PR adds a stylelint rule to enforce that.

I followed the implementation in LP for setting up a custom internal stylelint plugin.

#### Question:
It insists that `stylelint` and `dedent` now need to be prod dependencies instead of dev dependencies. Does anyone know why? 

### Test Plan:
- verify there are no stylelint errors
- add a tier 1 color CSS variable use somewhere in a component and verify it triggers the linter
